### DISCUSSION TITLE
Presentation Announcement: Techniques to debug Dotty issues

### DISCUSSION THREAD

**Discussion by anatoliykmetyuk** [scala/scala3#12088] [General Discussion]
Fixing Dotty issues is hard. One part of it is, given an issue, to understand why it is happening, what in the codebase causes it. And this task is complicated because the compiler is big and complex. The challenge here is navigation and awareness of what's going on in the code.

To my experience, having the right setup makes a huge difference when doing pretty much any task. Abraham Lincoln is quoted with the following:

> If I only had an hour to chop down a tree, I would spend the first 45 minutes sharpening my axe.” – Abraham Lincoln.

I spent some time thinking and playing with different setups to allow getting yourself to a root cause of an issue fast through the codebase. Now, I'd like to make a small presentation for those interested in what techniques help me navigate through the compiler with minimal overhead while tracing issues.

If you're interested to listen, here are the details:

When: Thursday 15th April, 15:00 CET
Where: https://epfl.zoom.us/j/85219469645?pwd=Y1dqWm9kejV6RHBRZGpuc0lHSUR4Zz09
How long: 1 hour

**Comment by SethTisue** [comment]
I hope this will be recorded? I don't relish waking up that early, here in Nevada :-)

**Reply by anatoliykmetyuk** [reply]
It will :) I'll post the recording here afterwards.

**Comment by anatoliykmetyuk** [comment]
Thank you everyone for coming to the talk! For those who didn't make it, the recording is available here: https://www.youtube.com/watch?v=H7-PYUHId0U

Slides: https://docs.google.com/presentation/d/1Vr-LulQWcbRwyiM4ZPhlc0sxZNiP7XyC6CE4jWoi0oE/edit?usp=sharing
6 minute demo: https://www.youtube.com/watch?v=1fOZXr2T0zI
